$OpenBSD: patch-tests_Makefile,v 1.3 2012/07/19 20:06:34 jasper Exp $

First chunk:
- Ensure the libraries needed by libregress.so can also be located and are preloaded.

--- tests/Makefile.orig	Thu Jul 19 22:00:55 2012
+++ tests/Makefile	Thu Jul 19 22:02:44 2012
@@ -41,7 +41,7 @@ clean :
 	rm -f $(REGRESS) $(REGRESS_OBJS) Regress-1.0.gir Regress-1.0.typelib
 
 check : all
-	cd .. && LD_LIBRARY_PATH=tests:$$LD_LIBRARY_PATH GI_TYPELIB_PATH=tests:$$GI_TYPELIB_PATH $(LUA) tests/test.lua
+	cd .. &&  LD_PRELOAD=/usr/lib/libpthread.so LD_LIBRARY_PATH=tests:$$LD_LIBRARY_PATH GI_TYPELIB_PATH=tests:$$GI_TYPELIB_PATH $(LUA) tests/test.lua
 
 $(REGRESS) : regress.o
 	$(CC) $(LDFLAGS) -o $@ regress.o $(LIBS)
@@ -58,7 +58,7 @@ Regress-1.0.gir : $(REGRESS)
 	  --namespace=Regress --nsversion=1.0				\
 	  --include=cairo-1.0 --include=Gio-2.0				\
 	  --library-path=/usr/lib --library-path=/usr/X11R6/lib		\
-	  --library-path=/usr/local/lib					\
+	  --library-path=${LOCALBASE}/lib				\
 	  $(GIDATADIR)/regress.c $(GIDATADIR)/regress.h			\
 	 -lregress
 
